rln/src/protocol.rs

```diff
@@ -338,13 +339,31 @@ pub fn compute_tree_root(
 
 // Generates a tupe (identity_secret, id_commitment) where
 // identity_secret is random and id_commitment = PoseidonHash(identity_secret)
+// RNG is instantiated using thread_rng()
 pub fn keygen() -> (Fr, Fr) {
     let mut rng = thread_rng();
     let identity_secret = Fr::rand(&mut rng);
     let id_commitment = poseidon_hash(&[identity_secret]);
     (identity_secret, id_commitment)
 }
 
+// Generates a tupe (identity_secret, id_commitment) where
+// identity_secret is random and id_commitment = PoseidonHash(identity_secret)
+// RNG is instantiated using 20 rounds of ChaCha seeded with the hash of the input
+pub fn seeded_keygen(signal: &[u8]) -> (Fr, Fr) {
```

Comment:
this function can accept a 1 byte signal, which I assume, is not secure.

I'd suggest to fix the signal size (and I am guessing that it should be 32 bytes).

Reply:
Fair point.